gdk4/src/functions.rs

```diff
@@ -1,12 +1,73 @@
 // Take a look at the license at the top of the repository in the LICENSE file.
 
-use crate::{ContentDeserializer, ContentSerializer};
+use crate::{ContentDeserializer, ContentSerializer, Event};
 use glib::object::IsA;
 use glib::translate::*;
 use std::future;
+use std::mem;
 use std::pin::Pin;
 use std::ptr;
 
+#[doc(alias = "gdk_events_get_angle")]
+pub fn events_get_angle(event1: &Event, event2: &Event) -> Option<f64> {
```

Comment:
Can the changes to the code be automated somehow? :) or should I just implement whatever allows me to have IsA<T> for my events?

Reply:
You can probably go via `Deref` here or not? And then implement this function only at the most generic type that has it

Reply:
What I meant here is the change of code from `
pub fn events_get_angle<P: IsA<Event>, Q: IsA<Event>>(event1: &P event2: &Q) -> Option<f64>;` to `pub fn events_get_angle(event1: &Event, event2: &Event) -> Option<f64>;`

There are more occurrences in GTK hence my question to avoid doing more stuff manually if possible

Reply:
Is that valid to call on any event types?

Reply:
In some functions like CellArea::event<P: IsA<Event>> yes; but some of them have to be manually done to use the right event type

Reply:
There are only 16 of them,  should be pretty doable, but would still love to know if there's a way to make a function accept any event in such case without having to downcast your events manually

Reply:
You'd have to introduce a new trait for that, or work with `AsRef<Event>` or similar

Reply:
I ended up working with AsRef